### PR TITLE
virtctl, pause: Remove globals

### PR DESCRIPTION
**Pull request by oshoval**
### What this PR does
Remove globals usage.
Ran unit tests locally (which do cover dryRun).

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
None
```

### CODE REVIEW

File: pkg/virtctl/pause/pause.go [hunk 1]
```diff
@@ -46,28 +46,33 @@ const (
...
 
 func NewPauseCommand(clientConfig clientcmd.ClientConfig) *cobra.Command {
 	cmd := &cobra.Command{
 		Use:   "pause vm|vmi (VM)|(VMI)",
 		Short: "Pause a virtual machine",
 		Long: `Pauses a virtual machine by freezing it. Machine state is kept in memory.
-First argument is the resource type, possible types are (case insensitive, both singular and plural forms) virtualmachineinstance (vmi) or virtualmachine (vm).
-Second argument is the name of the resource.`,
+		First argument is the resource type, possible types are (case insensitive, both singular and plural forms) virtualmachineinstance (vmi) or virtualmachine (vm).
+		Second argument is the name of the resource.`,
```
Comment (oshoval) [inline pkg/virtctl/pause/pause.go:61]: will revert this if desired

File: pkg/virtctl/pause/pause.go [hunk 2]
```diff
@@ -59,15 +61,18 @@ First argument is the resource type, possible types are (case insensitive, both
...
-				command:      COMMAND_PAUSE,
-				clientConfig: clientConfig,
-			}
-			return c.Run(args)
-		},
 	}
+
+	var dryRun bool
 	cmd.Flags().BoolVar(&dryRun, "dry-run", false, "--dry-run=false: Flag used to set whether to perform a dry run or not. If true the command will be executed without performing any changes.")
+	cmd.RunE = func(cmd *cobra.Command, args []string) error {
```
Comment (0xFelix) [inline pkg/virtctl/pause/pause.go:68]: Can you do it e.g. like here https://github.com/kubevirt/kubevirt/blob/82779585515c4df2945dea8980ee86c7749f6d93/pkg/virtctl/create/vm/vm.go#L274?

Use the structs members for the Flags?

Comment (oshoval) [reply pkg/virtctl/pause/pause.go]: Done

**Comment by oshoval** [comment]
Unexported struct as well

**Review by 0xFelix — COMMENTED** [review]
/approve

Thanks!

**Comment by oshoval** [comment]
can you also lgtm please ?

**Review by 0xFelix — APPROVED** [review]

**Pull request merged**